### ISSUE TITLE
Jetpack Connection: identify external links for screen readers

### ISSUE THREAD

**Issue by monsieur-z** [Automattic/jetpack#35479]
External links in the Jetpack connection flow are not identified as such by SRs (it’s the case in the footer, too).

Issue reported in p1HpG7-qVW-p2.

<img width="360" alt="Screenshot 2024-02-13 at 2 15 57 PM" src="https://github.com/Automattic/jetpack/assets/1620183/acb45d39-475f-48b2-93fd-7ddb6c8b2d0d">
<img width="1095" alt="Screenshot 2024-02-13 at 2 16 53 PM" src="https://github.com/Automattic/jetpack/assets/1620183/87136790-6860-4533-8bf1-9914840519f1">

---

1. The bare minimum is to ensure screen readers can identify external links. Following [this article](https://css-tricks.com/accessible-svgs/), this could probably be done by removing the `aria-hidden` attribute from the `svg`, adding a `role=img` attribute, and adding a title with the text screen readers should read.
<img width="344" alt="Screenshot 2024-02-13 at 2 18 58 PM" src="https://github.com/Automattic/jetpack/assets/1620183/949f8116-5c60-4b9b-93fb-fc8a0aa140b4">

2. If some of the links use the `ExternalLink` Gutenberg component, here's [an example of an accessible version](https://automattic.github.io/jetpack-storybook/?path=/docs/gutenberg_components-externallink--docs).

3. A good accessibility practice is to let users tailor their browsing experience, for instance, by not preventing zooming or by not preventing users from setting a font size. In the same logic, users should be able to decide when to open links in a new tab. By default, they have this choice. If we force links to open in a new tab, we remove this choice from them. There are only a few good reasons to do it, such as not wanting to lose the progress of a form. [This article](https://blog.pope.tech/2024/01/02/how-to-make-external-links-accessible/) explains it in more detail.
